src/math/history.h

```diff
@@ -26,7 +28,14 @@ template <class T> class History
             history_.erase(history_.begin());
 
         // Perform averaging of the datasets that we have
-        T averaged;
+        T averaged = [&]()
+        {
+            if (initialiser)
+                return initialiser();
+            else
+                return T();
+        }();
+
```

Comment:
Not sure why this is written inside a lambda?

```suggestion
        T averaged = initialiser ? initialiser() : T();

```